src/NetMQ/zmq/SocketBase.cs

```diff
@@ -768,17 +758,17 @@ public void Send(ref Msg msg, SendReceiveOptions flags)
                 ProcessCommands(timeout, false);
 
                 isMessageSent = XSend(ref msg, flags);
+                
                 if (isMessageSent)
                     break;
 
-                if (timeout > 0)
-                {
-                    timeout = (int)(end - Clock.NowMs());
-                    if (timeout <= 0)
-                    {
-                        throw new AgainException(innerException: null, message: "SocketBase.Send failed and timeout <= 0");
-                    }
-                }
+                if (timeout <= 0)
+                    continue;
+
+                timeout = (int)(end - Clock.NowMs());
+                    
+                if (timeout <= 0)
+                    throw new AgainException(innerException: null, message: "SocketBase.Send failed and timeout <= 0");
```

Comment:
once we will change the API to Try semantic we can also return false instead of throwing exception.